src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.html

```diff
@@ -1,260 +1,206 @@
-                 for="username"
-                 i18n>Username</label>
-          <div class="cd-col-form-input">
-            <input class="form-control"
-                   type="text"
-                   placeholder="Username..."
-                   id="username"
-                   name="username"
-                   formControlName="username"
-                   autocomplete="off"
-                   autofocus
-                   ngbTooltip="White spaces at the beginning and end will be trimmed"
-                   i18n-ngbTooltip
-                   cdTrim>
-            <span class="invalid-feedback"
-                  *ngIf="userForm.showError('username', formDir, 'required')"
-                  i18n>This field is required.</span>
-            <span class="invalid-feedback"
-                  *ngIf="userForm.showError('username', formDir, 'notUnique')"
-                  i18n>The username already exists.</span>
-          </div>
-        </div>
+<div cdsCol
+     [columnNumbers]="{md: 4}">
+  <ng-container *cdFormLoading="loading">
+    <form #frm="ngForm"
+          #formDir="ngForm"
+          [formGroup]="userForm"
+          novalidate>
+      <div i18n="form title|Example: Create Volume@@formTitle"
+           class="form-header">{{ action | titlecase }} {{ resource | upperFirst }}
+      </div>
+      <!-- UserName -->
+      <div class="form-item">
+        <cds-text-label for="username"
+                        cdRequiredField="Username"
+                        [invalid]="!userForm.controls.username.valid && userForm.controls.username.dirty"
+                        [invalidText]="usernameError"
+                        i18n>Username
+          <input cdsText
+                 placeholder="Username..."
+                 i18n-placeholder
+                 id="username"
+                 name="username"
+                 formControlName="username"
+                 [invalid]="!userForm.controls.username.valid && userForm.controls.username.dirty"
+                 autofocus>
+        </cds-text-label>
+        <ng-template #usernameError>
+          <span *ngIf="userForm.showError('username', formDir, 'required')">
+            <ng-container i18n>
+             This field is required.
+            </ng-container>
+          </span>
+          <span *ngIf="userForm.showError('username', formDir, 'notUnique')">
+            <ng-container i18n>
+              The username already exists.
+            </ng-container>
+          </span>
+        </ng-template>
+      </div>
+      <!-- Password -->
+      <div class="form-item">
+        <cds-password-label for="password"
+                            label="Password..."
+                            [invalid]="!userForm.controls.password.valid && userForm.controls.password.dirty"
+                            [invalidText]="passwordError"
+                            [helperText]="passwordhelperText"
+                            i18n>
```

Comment:
the label is missing in the form. Also this is a required field so make sure the label has the `(required)` marked.
![Screenshot from 2024-09-18 09-59-23](https://github.com/user-attachments/assets/1709e1a3-a24c-458d-b30e-469fc3702be8)

Reply:
and btw, the rule always gets shown even if I entered a valid password. Maybe we can hide it and show only when the password entered doesn't match with the rule?

Reply:
![image](https://github.com/user-attachments/assets/ecdc8664-0ca7-447d-aca0-7b1e76fe35fc)